src/tools/compiletest/src/common.rs

```diff
@@ -55,6 +55,7 @@ impl TestMode {
 
 // Note that coverage tests use the same test files for multiple test modes.
 string_enum! {
+    no_from_str,
     #[derive(Clone, Copy, PartialEq, Debug)]
     pub enum TestSuite {
         Assembly => "assembly",
```

Comment:
Suggestion [EXPLICIT 2/2]: so I'd prefer to see the "dumb" version of

```rs
AssemblyLlvm => "assembly-llvm",
AssemblyGcc => "assembly-gcc",
AssemblyCranelift => "assembly-cranelift",
```

which is also IMO easier to search for test suite specific behavior.

Reply:
Originally I changed like this:

`Assembly` -> `Assembly(CodegenBackend)`

However, it didn't work out well with the current codebase. If you prefer to have explicit names, find by me. :)